### PR TITLE
Add 2024-06-01-preview for redis enterprise

### PR DESCRIPTION
**Pull request by revanthballa1188**
## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by VeryEarly** [comment]
please record tests after API upgrade

**Comment by revanthballa1188** [comment]
Test recordings have been added

**Comment by dolauli** [comment]
@revanthballa1188 Please fix issues in examples that lead to the CI failure.

File: src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/examples/Invoke-AzRedisEnterpriseCacheForceDatabaseLinkToReplicationGroup.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+### Example 1: {{ Add title here }}
```
Comment (msJinLei) [inline src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/examples/Invoke-AzRedisEnterpriseCacheForceDatabaseLinkToReplicationGroup.md:1]: Please finish the examples and regenerated the module

File: src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/examples/Update-AzRedisEnterpriseCacheDatabaseDbRedisVersion.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+### Example 1: {{ Add title here }}
```
Comment (msJinLei) [inline src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/examples/Update-AzRedisEnterpriseCacheDatabaseDbRedisVersion.md:1]: Please finish the examples and regenerated the module

**Review by msJinLei — CHANGES_REQUESTED** [review]

File: src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/examples/Invoke-AzRedisEnterpriseCacheForceDatabaseLinkToReplicationGroup.md (hunk 1 above)
Comment (revanthballa1188) [reply src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/examples/Invoke-AzRedisEnterpriseCacheForceDatabaseLinkToReplicationGroup.md:1]: Added

File: src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/examples/Update-AzRedisEnterpriseCacheDatabaseDbRedisVersion.md (hunk 2 above)
Comment (revanthballa1188) [reply src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/examples/Update-AzRedisEnterpriseCacheDatabaseDbRedisVersion.md:1]: Added

File: src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/examples/Invoke-AzRedisEnterpriseCacheForceDatabaseLinkToReplicationGroup.md (hunk 1 above)
Comment (msJinLei) [reply src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/examples/Invoke-AzRedisEnterpriseCacheForceDatabaseLinkToReplicationGroup.md:1]: @revanthballa1188 You should add the example here and the files of `docs` are all generated.

File: src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/examples/Update-AzRedisEnterpriseCacheDatabaseDbRedisVersion.md (hunk 2 above)
Comment (msJinLei) [reply src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/examples/Update-AzRedisEnterpriseCacheDatabaseDbRedisVersion.md:1]: You should add the example here and the files of docs are all generated.

File: src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/examples/Invoke-AzRedisEnterpriseCacheForceDatabaseLinkToReplicationGroup.md (hunk 1 above)
Comment (revanthballa1188) [reply src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/examples/Invoke-AzRedisEnterpriseCacheForceDatabaseLinkToReplicationGroup.md]: Got it, Updated now!

File: src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/examples/Update-AzRedisEnterpriseCacheDatabaseDbRedisVersion.md (hunk 2 above)
Comment (revanthballa1188) [reply src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/examples/Update-AzRedisEnterpriseCacheDatabaseDbRedisVersion.md]: Got it! updated now!

File: src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/docs/Invoke-AzRedisEnterpriseCacheDatabaseFlush.md [hunk 3]
```diff
@@ -14,16 +14,16 @@ Flushes all the keys in this database and also from its linked databases.
 
 ### FlushExpanded (Default)
 ```
-Invoke-AzRedisEnterpriseCacheDatabaseFlush -ClusterName <String> -ResourceGroupName <String> [-Name <String>]
```
Comment (msJinLei) [inline src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/docs/Invoke-AzRedisEnterpriseCacheDatabaseFlush.md:17]: -Name is removed

File: src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/docs/Invoke-AzRedisEnterpriseCacheForceDatabaseUnlink.md [hunk 4]
```diff
@@ -83,23 +83,9 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
-### -DatabaseName
```
Comment (msJinLei) [inline src/RedisEnterpriseCache/RedisEnterpriseCache.Autorest/docs/Invoke-AzRedisEnterpriseCacheForceDatabaseUnlink.md:86]: Remove DatabaseName

**Review by msJinLei — CHANGES_REQUESTED** [review]
I find several breaking changes in the PR including removing existing parameters and the change of namespaces. But why static analysis passes? We need check again

**Review by msJinLei — APPROVED** [review]

**Comment by msJinLei** [comment]
@revanthballa1188 Please provide the change log for this PR

**Pull request merged**